server/internal/usecase/interactor/request.go

```diff
@@ -159,6 +159,19 @@ func (r Request) Update(ctx context.Context, param interfaces.UpdateRequestParam
 			req.SetReviewers(param.Reviewers)
 		}
 		if param.Items != nil {
+			repoItems, err := r.repos.Item.FindByIDs(ctx, param.Items.IDs())
+			if err != nil {
+				return nil, err
+			}
+
+			for _, item := range repoItems {
+				if version.MatchVersionOrRef(item.Version().OrRef(), nil,
+					func(r version.Ref) bool {
+						return r == version.Public
+					}) {
+					return nil, errors.New("already published")
+				}
+			}
```

Comment:
This code does no meaningful processing; `item.Version()` always returns `Version` not `Ref`, so the function passed to MatchVersionOrRef is never executed.

```suggestion
			repoItems, err := r.repos.Item.FindByIDs(ctx, param.Items.IDs(), version.Public.OrVersion().Ref())
			if err != nil {
				return nil, err
			}

			for _, item := range repoItems {
				if item.Refs().Has(version.Latest) {
					return nil, errors.New("already published")
				}
			}
```

Note that extend `Item.FindByIDs` method arguments to accept `*version.VersionOrRef` (if it is nil, the method should find latest versions)